### PR TITLE
feat(ivy): bridge component styles into the component renderer

### PR DESCRIPTION
**Pull request by matsko**

### CODE REVIEW

File: packages/core/src/render3/definition.ts [hunk 1]
```diff
@@ -210,11 +211,41 @@ export function defineComponent<T>(componentDefinition: {
...
+   * - `ViewEncapsulation.None`: Use global CSS without any
+   * encapsulation.
+   *
+   * If not supplied, the value is taken from `CompilerOptions`. The default compiler option is
+   * `ViewEncapsulation.Emulated`.
+   *
+   * If the policy is set to `ViewEncapsulation.Emulated` and the component has no `styles`
+   * or `styleUrls` specified, the policy is automatically switched to `ViewEncapsulation.None`.
+   */
+  encapsulation?: ViewEncapsulation;
```
Comment (trotyl) [inline packages/core/src/render3/definition.ts]: Should new `ViewEncapsulation.ShadowDom` option be considered now?

File: packages/core/test/bundling/todo_r2/index.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  }
+}
+
+@NgModule({
+  declarations: [ToDoAppComponent],
+  imports: [CommonModule, BrowserModule],
+  bootstrap: [ToDoAppComponent]
+})
+class ToDoAppModule {
+  ngDoBootstrap(app: any) { app.bootstrap(ToDoAppComponent); }
```
Comment (trotyl) [inline packages/core/test/bundling/todo_r2/index.ts]: Shouldn't only one of `ngDoBootstrap` method or `bootstrap` array be used?

File: packages/core/src/render3/definition.ts (hunk 1 above)
Comment (matsko) [reply packages/core/src/render3/definition.ts]: Ah yes right ... there isn't a comment description for that one.

File: packages/core/test/bundling/todo_r2/index.ts (hunk 2 above)
Comment (matsko) [reply packages/core/test/bundling/todo_r2/index.ts]: Fixed. Thank you.

File: packages/core/test/bundling/hello_world/bundle.golden_symbols.json [hunk 3]
```diff
@@ -134,6 +134,9 @@
   {
     "name": "createLViewData"
   },
+  {
+    "name": "createRendererType2"
```
Comment (mhevery) [inline packages/core/test/bundling/hello_world/bundle.golden_symbols.json]: revert: see comment above

File: packages/core/test/bundling/todo/bundle.golden_symbols.json [hunk 4]
```diff
@@ -401,6 +401,9 @@
   {
     "name": "createOutput$1"
   },
+  {
+    "name": "createRendererType2"
```
Comment (mhevery) [inline packages/core/test/bundling/todo/bundle.golden_symbols.json]: revert: see comment above

File: packages/core/test/bundling/todo_r2/OUTSTANDING_WORK.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# Outstanding on the `Todo` app
```
Comment (mhevery) [inline packages/core/test/bundling/todo_r2/OUTSTANDING_WORK.md]: revert this file

File: packages/core/test/bundling/todo_r2/index.ts [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+  toggleCompletion(todo: Todo) { todo.completed = !todo.completed; }
+
+  remove(todo: Todo) { this.todos.splice(this.todos.indexOf(todo), 1); }
+
+  add(title: string) { this.todos.push(new Todo(title)); }
+}
+
+@Component({
+  selector: 'todo-app',
+  // TODO(misko): make this work with `[(ngModel)]`
```
Comment (mhevery) [inline packages/core/test/bundling/todo_r2/index.ts]: for the non `r2` version of the todo, we should set `encapsulation: ViewEncapsulation.None`

**Review by mhevery — CHANGES_REQUESTED** [review]
Minor fixes in 6fd2a4304f

**Review by mhevery — APPROVED** [review]

**Comment by matsko** [comment]
http://test/ui#id=OCL:207597856:BASE:209984617:1535054543951:da5d01d6

**Pull request closed without merging**